// ais-stream-decoder/dist/messages/ais-message-8-inland-static.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ais_message_1 = __importDefault(require("./ais-message"));
const format_1 = __importDefault(require("../format"));
class AisMessage8InlandStatic extends ais_message_1.default {
    // eslint-disable-next-line max-statements
    constructor(messageType, channel, bitField) {
        super(messageType, channel, bitField);
        this.dac = bitField.getInt(40, 10);
        this.fid = bitField.getInt(50, 6);
        this.eni = bitField.getString(56, 48);
        this.length = format_1.default.inlandLengthOrBeam(bitField.getInt(104, 13));
        this.beam = format_1.default.inlandLengthOrBeam(bitField.getInt(117, 10));
        this.shipType = bitField.getInt(127, 14);
        this.hazardCode = bitField.getInt(141, 3);
        this.draught = format_1.default.inlandDraught(bitField.getInt(144, 11));
        this.loadStatus = bitField.getInt(155, 2);
        this.speedQuality = bitField.getBoolean(157, 1);
        this.courseQuality = bitField.getBoolean(158, 1);
        this.headingQuality = bitField.getBoolean(159, 1);
    }
}
exports.default = AisMessage8InlandStatic;
